modules/dxp/apps/osb/osb-faro/osb-faro-service/src/main/java/com/liferay/osb/faro/internal/model/listener/UserModelListener.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.osb.faro.internal.model.listener;
+
+import com.liferay.osb.faro.model.FaroProject;
+import com.liferay.osb.faro.model.FaroUser;
+import com.liferay.osb.faro.service.FaroProjectLocalService;
+import com.liferay.osb.faro.service.FaroUserLocalService;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.model.User;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sillas Cavalcanti
+ */
+@Component(service = ModelListener.class)
+public class UserModelListener extends BaseModelListener<User> {
+
+	@Override
+	public void onAfterRemove(User user) {
+		List<FaroProject> faroProjects =
+			_faroProjectLocalService.getFaroProjects(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		for (FaroProject project : faroProjects) {
+			FaroUser faroUser = _faroUserLocalService.fetchFaroUser(
+				project.getGroupId(), user.getEmailAddress());
+
+			_faroUserLocalService.deleteFaroUser(faroUser);
```

Comment:
@sillascavalcanti, It works good, but in order to have a better performance, can you try to do this in batch? You can send a list of faroUsers as parameter and delete in LocalService layer